From e38ebe4d4d3739c95245bf8077a54754d72ec95c Mon Sep 17 00:00:00 2001
From: Chris Renshaw <osm0sis@outlook.com>
Date: Mon, 12 Sep 2016 16:55:01 -0300
Subject: [PATCH 22/32] depmod: fix syntax with modules in parameter

Patch by Tanguy Pruvot <tanguy.pruvot@gmail.com>
Change-Id: I21b8664db01cf0132db82f8d6caa1a0e77e71004
from https://github.com/tpruvot/android_external_busybox
Rebased for busybox 1.25.0 by Chris Renshaw <osm0sis@outlook.com>
---
 modutils/depmod.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/modutils/depmod.c b/modutils/depmod.c
index bb42bbefe..5e4c4c538 100644
--- a/modutils/depmod.c
+++ b/modutils/depmod.c
@@ -49,7 +49,10 @@ static int FAST_FUNC parse_module(struct recursive_state *state,
 	image = xmalloc_open_zipped_read_close(fname, &len);
 
 	e = moddb_get_or_create(modules, bb_get_last_path_component_nostrip(fname));
-	e->name = xstrdup(fname + 2); /* skip "./" */
+	if (strncmp(fname, "./", 2) == 0)
+		e->name = xstrdup(fname + 2);
+	else
+		e->name = xstrdup(fname);
 
 	for (ptr = image; ptr < image + len - 10; ptr++) {
 		if (is_prefixed_with(ptr, "depends=")) {
-- 
2.31.1

